app/models/client_opportunity_match.rb

```diff
@@ -785,7 +756,7 @@ def already_active_for_opportunity
 
   def init_referral_event
     return unless Warehouse::Base.enabled?
-    return referral_event if referral_event.present?
+    return active_referral_event if active_referral_event.present?
     return unless project_client&.from_hmis?
 
     create_referral_event(
```

Comment:
```suggestion
    create_active_referral_event(
```

Reply:
I don't think that's necessary.  `active_referral_event` is a `has_one` with a scope, `referral_events` is a `has_many`.  Because it's creating a new one, the `referral_result` will be `nil` when just creating and event, and I have a feeling the scope on the `has_one` may be problematic.

Reply:
You were correct on this one.  Not sure why it was working, but it's been fixed now.